File: argentum-api/api/tests/data/statuses.py
from api.models.status import Status
from api.tests.utils.test_objects import TestObjects


class TestStatuses(TestObjects):
    MODEL = Status

    PAID: MODEL
    PENDING: MODEL

    STAFF: MODEL
    PENDING_PATCHED: MODEL

    @classmethod
    def init(cls):
        cls.PAID = cls.MODEL(
            id=16010,
            internal_name='paid',
            display_name='Paid',
            color='#00ff00'
        )

        cls.PENDING = cls.MODEL(
            id=16020,
            internal_name='pending',
            display_name='Pending',
            color='#ff0000'
        )

        cls.SAVED = [cls.PAID, cls.PENDING]

        cls.STAFF = cls.MODEL(
            id=16030,
            internal_name='staff',
            display_name='Staff',
            color='#0000ff',
        )

        cls.PENDING_PATCHED = cls.MODEL(
            id=16021,
            internal_name='pen',
            display_name='Pending',
            color='#ff0000'
        )

        cls.UNSAVED = [cls.STAFF, cls.PENDING_PATCHED]

File: argentum-ui/src/app/common/rest-service/test-data/statuses.ts
import {Status} from '../../model/status';

export namespace Statuses {
  export const PAID = new Status(16010, 'paid', 'Paid', '#00ff00');
  export const PENDING = new Status(16020, 'pending', 'Pending', '#ff0000');

  export const ALL = [PAID, PENDING];

  export const STAFF = new Status(16030, 'staff', 'Staff', '#0000ff');
  export const PENDING_PATCHED_REQUEST = new Status(16021, 'pen', undefined, undefined);
  export const PENDING_PATCHED_RESPONSE = new Status(16021, 'pen', 'Pending', '#ff0000');
}
